queue_services/entity-emailer/src/entity_emailer/email_processors/incorp_notification.py

```diff
@@ -12,13 +12,166 @@
+                    'fileBytes': receipt_encoded.decode('utf-8'),
+                    'fileUrl': '',
+                    'attachOrder': '2'
+                }
+            )
+    if stage == 'registered':
+        noa = requests.get(
+            f'{current_app.config.get("LEGAL_API_URL")}/businesses/{business["identifier"]}/filings/{filing.id}\
+            ?type=noa',
+            headers=headers
+        )
+        if noa.status_code != HTTPStatus.CREATED:
+            logger.error('Failed to get noa pdf for filing: %s', filing.id)
+            capture_message(f'Email Queue: filing id={filing.id}, error=noa generation', level='error')
+        else:
+            noa_encoded = base64.b64encode(noa.content)
+            pdfs.append(
+                {
+                    'fileName': 'Notice of Articles.pdf',
+                    'fileBytes': noa_encoded.decode('utf-8'),
+                    'fileUrl': '',
+                    'attachOrder': '1'
+                }
+            )
+        certificate = requests.get(
+            f'{current_app.config.get("LEGAL_API_URL")}/businesses/{business["identifier"]}/filings/{filing.id}\
+            ?type=certificate',
+            headers=headers
+        )
+        if certificate.status_code != HTTPStatus.CREATED:
+            logger.error('Failed to get certificate pdf for filing: %s', filing.id)
+            capture_message(f'Email Queue: filing id={filing.id}, error=certificate generation', level='error')
+        else:
+            certificate_encoded = base64.b64encode(certificate.content)
+            pdfs.append(
+                {
+                    'fileName': 'Incorporation Certificate.pdf',
+                    'fileBytes': certificate_encoded.decode('utf-8'),
+                    'fileUrl': '',
+                    'attachOrder': '2'
+                }
+            )
 
+    return pdfs
 
-def process(email_msg):
+
+def _get_recipients(stage: str, filing_json: dict):
+    """Get the recipients for the incorporation output."""
+    recipients = filing_json['filing']['incorporationApplication']['contactPoint']['email']
+    if stage == 'filed':
+        parties = filing_json['filing']['incorporationApplication'].get('parties')
+        comp_party_email = None
+        for party in parties:
+            for role in party['roles']:
+                if role['roleType'] == 'Completing Party':
+                    comp_party_email = party['officer']['email']
+                    break
+        recipients = f'{recipients}, {comp_party_email}'
+    return recipients
+
+
+def process(email_msg: dict, token: str):
     """Build the email for Business Number notification."""
     logger.debug('incorp_notification: %s', email_msg)
-    email = 'incorp hardcoded'
-    # build email from template
-    # get links etc.
-    return email
+    # get template and fill in parts
+    template = Path(f'email_templates/BC-{email_msg["option"]}-success.html').read_text()
```

Comment:
I'm not sure if this filename pattern will work in the future.
- "BC" was for BComp; this assumed other types would have different templates; will they?
- "BC-MRAS" is another template (near future), that already doesn't follow this pattern

Thoughts on filename pattern?

Reply:
I think we can change that as we add in more templates - since this is within the incorporation processor it won't change until we add in other corps. MRAS will have it's own processor

Reply:
> @kialj876 Where is the email subject set? The requirements doc specifies a few different subjects depending on scenario.
> 
> https://docs.google.com/document/d/1HZr3sIHYpdakG49Y6RuP2rXTFxEtSssIp0JvWrfL6yc/edit

ah you're right - I assumed they were all the same. I have a feeling that will change

Reply:
OK for now. Change as needed in future!